service/test/ConstConfigIOGroupTest.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "ConstConfigIOGroup.hpp"
+
+#include <set>
+#include <string>
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <unistd.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+#include "geopm_topo.h"  // for geopm_domain_e
+#include "geopm_test.hpp" // for GEOPM_EXPECT_THROW_MESSAGE
+
+#include "geopm/Exception.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+
+using geopm::ConstConfigIOGroup;
+using geopm::Exception;
+using geopm::PlatformIO;
+using geopm::PlatformTopo;
+
+class ConstConfigIOGroupTest : public::testing::Test
+{
+    protected:
+        void SetUp() override;
+        void TearDown() override;
+};
+
+void ConstConfigIOGroupTest::SetUp()
+{
+}
+
+TEST_F(ConstConfigIOGroupTest, input_empty_string)
+{
+    GEOPM_EXPECT_THROW_MESSAGE(
+        ConstConfigIOGroup iogroup(""),
+        GEOPM_ERROR_INVALID,
+        "ConstConfigIOGroup::parse_config_json(): "
+        "detected a malformed JSON string"
+    );
+}
+
+TEST_F(ConstConfigIOGroupTest, input_gibberish)
```

Comment:
This is a good start for the negative tests. In case you don't have these in mind, make sure you cover cases like:
- Signals with missing required fields
- Fields with values of an incorrect type
- Fields with a value outside of their domain/enum
- Duplicate signals
- A signal with an empty "values" array